File: utils/src/main/java/com/yujing/utils/YRunnable.java
package com.yujing.utils;

import com.yujing.contract.YRun;

/**
 * Runnable线程，里面包含try，解决Runnable.run包含try过于复杂问题
 */
public class YRunnable implements Runnable {
    private final YRun yRun;

    public YRunnable(YRun yRun) {
        this.yRun = yRun;
    }

    @Override
    public void run() {
        if (yRun != null) {
            try {
                yRun.run();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }
}

File: utils/src/main/java/com/yujing/contract/YRunnable.kt
package com.yujing.contract

/**
 * 给runnable外加了Exception注解
 */
/*
用法：
解决Thread.sleep必须try
Thread t  = new Thread((YRunnable) () -> {
    Thread.sleep(1000);
});
t.start();
 */
interface YRunnable : Runnable {
    @Throws(Throwable::class)
    override fun run()
}